projects/adrv9026/common/adrv9026_bd.tcl

```diff
@@ -0,0 +1,228 @@
+  SAMPLE_DATA_WIDTH $TX_SAMPLE_WIDTH \
+]
+
+adi_tpl_jesd204_tx_create tx_adrv9026_tpl_core $TX_NUM_OF_LANES \
+                                               $TX_NUM_OF_CONVERTERS \
+                                               $TX_SAMPLES_PER_FRAME \
+                                               $TX_SAMPLE_WIDTH
+
+ad_ip_instance axi_dmac axi_adrv9026_tx_dma
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.DMA_TYPE_SRC 0
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.DMA_TYPE_DEST 1
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.CYCLIC 1
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.ASYNC_CLK_DEST_REQ 1
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.ASYNC_CLK_SRC_DEST 1
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.ASYNC_CLK_REQ_SRC 1
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.DMA_DATA_WIDTH_DEST $dac_dma_data_width
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.MAX_BYTES_PER_BURST 256
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.DMA_DATA_WIDTH_SRC 128
+ad_ip_parameter axi_adrv9026_tx_dma CONFIG.FIFO_SIZE 32
+
+ad_dacfifo_create $dac_fifo_name $dac_data_width $dac_dma_data_width $dac_fifo_address_width
+
+# adc peripherals
+
+ad_ip_instance axi_adxcvr axi_adrv9026_rx_xcvr
+ad_ip_parameter axi_adrv9026_rx_xcvr CONFIG.NUM_OF_LANES $RX_NUM_OF_LANES
+ad_ip_parameter axi_adrv9026_rx_xcvr CONFIG.QPLL_ENABLE 0
+ad_ip_parameter axi_adrv9026_rx_xcvr CONFIG.TX_OR_RX_N 0
+ad_ip_parameter axi_adrv9026_rx_xcvr CONFIG.SYS_CLK_SEL 0
+ad_ip_parameter axi_adrv9026_rx_xcvr CONFIG.OUT_CLK_SEL 3
+
+adi_axi_jesd204_rx_create axi_adrv9026_rx_jesd $RX_NUM_OF_LANES
+ad_ip_parameter axi_adrv9026_rx_jesd/rx CONFIG.SYSREF_IOB {false}
+
+ad_ip_instance util_cpack2 util_adrv9026_rx_cpack [list \
+  NUM_OF_CHANNELS $RX_NUM_OF_CONVERTERS \
+  SAMPLES_PER_CHANNEL $RX_SAMPLES_PER_CHANNEL \
+  SAMPLE_DATA_WIDTH $RX_SAMPLE_WIDTH \
+  ]
+
+adi_tpl_jesd204_rx_create rx_adrv9026_tpl_core $RX_NUM_OF_LANES \
+                                               $RX_NUM_OF_CONVERTERS \
+                                               $RX_SAMPLES_PER_FRAME \
+                                               $RX_SAMPLE_WIDTH
+
+ad_ip_instance axi_dmac axi_adrv9026_rx_dma
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.DMA_TYPE_SRC 2
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.SYNC_TRANSFER_START 1
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.ASYNC_CLK_DEST_REQ 1
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.ASYNC_CLK_SRC_DEST 1
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.ASYNC_CLK_REQ_SRC 1
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.DMA_DATA_WIDTH_SRC [expr 32*$RX_NUM_OF_LANES]
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.MAX_BYTES_PER_BURST 256
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.DMA_DATA_WIDTH_DEST 128
+ad_ip_parameter axi_adrv9026_rx_dma CONFIG.FIFO_SIZE 32
```

Comment:
Given that we don't have a FIFO in the path, can we increase this to the maximum size ?

Reply:
The maximum size is 32